x/feegrant/filtered_fee_test.go

```diff
@@ -188,3 +189,73 @@ func TestFilteredFeeValidAllow(t *testing.T) {
 		})
 	}
 }
+
+// invalidInterfaceAllowance does not implement proto.Message
+type invalidInterfaceAllowance struct {
+}
+
+// compilation time interface implementation check
+var _ feegrant.FeeAllowanceI = (*invalidInterfaceAllowance)(nil)
+
+func (i invalidInterfaceAllowance) Accept(ctx sdk.Context, fee sdk.Coins, msgs []sdk.Msg) (remove bool, err error) {
+	return false, nil
+}
+
+func (i invalidInterfaceAllowance) ValidateBasic() error {
+	return nil
+}
+
+// invalidProtoAllowance can not run proto.Marshal
+type invalidProtoAllowance struct {
+	invalidInterfaceAllowance
+}
+
+// compilation time interface implementation check
+var _ feegrant.FeeAllowanceI = (*invalidProtoAllowance)(nil)
+var _ proto.Message = (*invalidProtoAllowance)(nil)
+
+func (i invalidProtoAllowance) Reset() {
+}
+
+func (i invalidProtoAllowance) String() string {
+	return ""
+}
+
+func (i invalidProtoAllowance) ProtoMessage() {
+}
+
+func TestSetAllowance(t *testing.T) {
+	cases := map[string]struct {
+		allowance feegrant.FeeAllowanceI
+		valid     bool
+	}{
+		"valid allowance": {
+			allowance: &feegrant.BasicAllowance{},
+			valid:     true,
+		},
+		"invalid interface allowance": {
+			allowance: &invalidInterfaceAllowance{},
+			valid:     false,
+		},
+		"empty allowance": {
+			allowance: (*invalidProtoAllowance)(nil),
```

Comment:
Do you really need `invalidProtoAllowance`? Is this possible:
```suggestion
			allowance: (*feegrant.BasicAllowance)(nil),
```

Reply:
That change cause panic at [`proto.Marshal`](https://github.com/line/lbm-sdk/blob/d48d8a7094acb0253e0ba9b279f4ee5de3daef4a/codec/types/any.go#L68).
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
```